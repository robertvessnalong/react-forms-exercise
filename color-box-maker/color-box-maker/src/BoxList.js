import { useState } from 'react';
import NewBoxForm from './NewBoxForm';
import { v4 as uuidv4 } from 'uuid';
import Box from './Box';

const BoxList = () => {
  const INITAL_STATE = [];
  const [boxList, setBox] = useState(INITAL_STATE);
  const addBox = (newBox) => {
    setBox((boxList) => [...boxList, { ...newBox, id: uuidv4() }]);
  };
  const RemoveItem = (id) => {
    const updatedBoxList = boxList.filter((box) => box.id !== id);
    setBox(updatedBoxList);
  };

  return (
    <>
      <NewBoxForm addBox={addBox} />
      <div className='Boxlist'>
        {boxList.map((box) => (
          <Box
            remove={RemoveItem}
            height={box.height}
            width={box.width}
            bgclr={box.bgclr}
            key={box.id}
            id={box.id}
          />
        ))}
      </div>
    </>
  );
};

export default BoxList;
