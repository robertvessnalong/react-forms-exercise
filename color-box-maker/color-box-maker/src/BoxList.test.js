import { render, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

it('renders without crashing', () => {
  render(<BoxList />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it('add new item', () => {
  const { getByLabelText, queryByText, getByText } = render(<BoxList />);
  const width = getByLabelText('Width:');
  const height = getByLabelText('Height:');
  const color = getByLabelText('Background Color:');
  const btn = queryByText('Add Box');
  fireEvent.change(width, { target: { value: '100' } });
  fireEvent.change(height, { target: { value: '100' } });
  fireEvent.change(color, { target: { value: 'red' } });
  fireEvent.click(btn);
  const x = getByText('X');
  expect(x).toBeInTheDocument();
});
