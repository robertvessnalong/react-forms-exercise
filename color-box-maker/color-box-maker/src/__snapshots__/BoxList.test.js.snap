// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      name="height"
      placeholder="Height of Box"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      name="width"
      placeholder="Width of Box"
      type="text"
      value=""
    />
    <label
      for="bgclr"
    >
      Background Color:
    </label>
    <input
      id="bgclr"
      name="bgclr"
      placeholder="Background Color of Box"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
  <div
    class="Boxlist"
  />
</DocumentFragment>
`;
